cmd/baseserver/main.go

```diff
@@ -17,12 +22,59 @@ var srvCli = &cobra.Command{
 	Long:  `Baseserver presents  a nice (not raw hex) interface to the basecoin blockchain structure.`,
 }
 
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Serve the light REST client for tendermint",
+	Long:  "Access basecoin via REST",
+	RunE:  serve,
+}
+
+const (
+	envPortFlag = "port"
+	defaultAlgo = "ed25519"
+)
+
+func init() {
+	_ = serveCmd.PersistentFlags().Int(envPortFlag, 8998, "the port to run the server on")
+}
+
+func serve(cmd *cobra.Command, args []string) error {
+	router := mux.NewRouter()
+
+	routeRegistrars := []func(*mux.Router) error{
```

Comment:
I like the flexibility of this function very much. But still don't want to have to add all this code in every app. The only thing that changes between apps is routeRegistrars.

For DRY, I would prefer adding some global way to set the `routeRegistrars` from main.go (calling a function, directly setting a public variable, ...) but have the rest of this file back in `client/rest`